cmd/gardener-apiserver/app/gardener_apiserver.go

```diff
@@ -247,6 +248,11 @@ func (o *Options) config(kubeAPIServerConfig *rest.Config, kubeClient *kubernete
 
 // Run runs gardener-apiserver with the given Options.
 func (o *Options) Run(ctx context.Context) error {
+	logger := logger.NewLogger("", "")
+	logger.Info("Starting Gardener API server...")
+	logger.Infof("Version: %+v", version.Get())
+	logger.Infof("Feature Gates: %s", utilfeature.DefaultFeatureGate)
```

Comment:
If I start it with no feature gate flag set, it logs
```
{"level":"info","msg":"Feature Gates: ","time":"2021-09-21T18:33:19+03:00"}
```

Can it be made to print all feature gates and their enablement status? E.g. print `DisallowKubeconfigRotationForShootInDeletion=true` even when it is not explicitly enabled via CLI flag?

Reply:
On the other hand, I think GCM, `gardenlet` etc. would similarly not log any feature gates not explicitly enabled / disabled, so we change it for the apiserver, we should change it everywhere.

Reply:
Valid point. The reason is that the featureGate String implementation considers only explicitly enabled features and their values and does not consider known features and their default values.

Ref https://github.com/kubernetes/kubernetes/blob/092fbfbf53427de67cac1e9fa54aaa09a28371d7/staging/src/k8s.io/component-base/featuregate/feature_gate.go#L247-L255

I am even considering that the String() func can be implemented in such way because in K8s there are a lot of known features (more that 100) and the output itself would "explode" with such high amount of feature gates.

So that's why I personally wouldn't invest into adapting the upstream func.

So bottom line, we have to treat these logged feature gates as the ones that are explicitly configured for the component (and not as the ones that also contain the default known feature gates and their values - this is what is actually used by the `Enabled` func itself at runtime). 😅

Reply:
OK, then lets' leave this as it is.